lib/seadsa/DsaLocal.cc

```diff
@@ -468,36 +479,39 @@ class IntraBlockBuilder : public InstVisitor<IntraBlockBuilder>,
   void visitAllocWrapperCall(CallSite &CS);
   void visitAllocationFnCall(CallSite &CS);
 
-  /// Returns true if \p F is a \p seadsa_ family of functions
-  static bool isSeaDsaFn(const Function *fn) {
-    if (!fn)
-      return false;
-    auto n = fn->getName();
-    return n.startswith("sea_dsa_") || n.startswith("seadsa_");
-  }
-
-  static bool isSeaDsaAliasFn(const Function *fn) {
-    if (!fn)
-      return false;
-    auto name = fn->getName();
-    // due to name change from sea_dsa to seadsa, support both version
-    return name.equals("sea_dsa_alias") || name.equals("seadsa_alias");
-  }
+  SeadsaFn getSeaDsaFn (const Function *fn) {
+    if (!fn)  return SeadsaFn::UNKNOWN;
 
-  static bool isSeaDsaCollapseFn(const Function *fn) {
-    if (!fn)
-      return false;
-    auto name = fn->getName();
-    // due to name change from sea_dsa to seadsa, support both version
-    return name.equals("sea_dsa_collapse") || name.equals("seadsa_collapse");
+    return StringSwitch<SeadsaFn>(fn->getName())
+              .Case("sea_dsa_set_modified", SeadsaFn::MODIFY)
+              .Case("sea_dsa_set_read"    , SeadsaFn::READ)
+              .Case("sea_dsa_set_ptrtoint", SeadsaFn::PTR_TO_INT)
+              .Case("sea_dsa_alias"       , SeadsaFn::ALIAS)
+              .Case("sea_dsa_collapse"    , SeadsaFn::COLLAPSE)
+              .Case("sea_dsa_mk_seq"      , SeadsaFn::MAKE_SEQ)
+              .Default(SeadsaFn::UNKNOWN);
   }
 
-  static bool isSeaDsaMkSequenceFn(const Function *fn) {
+  /// Returns true if \p F is a \p seadsa_ family of functions
+  static bool isSeaDsaFn(const Function *fn) {
     if (!fn)
       return false;
-    auto name = fn->getName();
-    // due to name change from sea_dsa to seadsa, support both version
-    return name.equals("sea_dsa_mk_seq") || name.equals("seadsa_mk_seq");
+    auto n = fn->getName();
+    return n.startswith("sea_dsa_");
+  }
+
+  //return function pointer to seadsa::Node setter
+  std::function<seadsa::Node&(seadsa::Node*, bool)> getSeaDsaAttrbFunc(const Function *fn) {   
```

Comment:
with `using` to name the returned function the code will be easier to read.

Are you sure you are not missing a clang-format? The line looks pretty long.

Reply:
Fair enough, done at 3e18195